modules/cudaimgproc/src/cuda/hist.cu

```diff
@@ -105,6 +105,72 @@ namespace hist
         if (stream == 0)
             cudaSafeCall( cudaDeviceSynchronize() );
     }
+
+    __global__ void histogram256Kernel(const uchar* src, int cols, int rows, size_t step, const uchar* mask, int* hist)
+    {
+        __shared__ int shist[256];
+
+        const int y = blockIdx.x * blockDim.y + threadIdx.y;
+        const int tid = threadIdx.y * blockDim.x + threadIdx.x;
+
+        shist[tid] = 0;
+        __syncthreads();
+
+        if (y < rows)
+        {
+            const unsigned int* rowPtr = (const unsigned int*) (src + y * step);
+            const unsigned int* maskRowPtr = (const unsigned int*) (mask + y * step);
```

Comment:
This code assumes that `mask.step` is equal to `src.step`. This is not true in general and should be checked.